### PR TITLE
Fix #467: fix mantid on f36

### LINKED ISSUE #467: f36 import mantid fails
```
CMake (main)$ rsmantid -c 'from mantid import simpleapi'
Traceback (most recent call last):
  File "/home/vagrant/.local/lib/mantid/lib/mantid/utils/_utils.py", line 37, in import_mantid_cext
    lib = import_module(modulename, package)
  File "/home/vagrant/.pyenv/versions/3.9.15/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 666, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 565, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 1173, in create_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
ImportError: /home/vagrant/.local/lib/mantid/lib/mantid/api/../../../lib/libMantidGeometry.so: undefined symbol: _ZN24BRepBuilderAPI_MakeShape5BuildEv

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/vagrant/.local/lib/mantid/lib/mantid/utils/_utils.py", line 41, in import_mantid_cext
    lib = import_module(modulename.lstrip('.'))
  File "/home/vagrant/.pyenv/versions/3.9.15/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 984, in _find_and_load_unlocked
ModuleNotFoundError: No module named '_api'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/vagrant/.local/lib/mantid/lib/mantid/__init__.py", line 109, in <module>
    from mantid import api as _api  # noqa: F401, E402
  File "/home/vagrant/.local/lib/mantid/lib/mantid/api/__init__.py", line 20, in <module>
    import_mantid_cext('._api', 'mantid.api', globals())
  File "/home/vagrant/.local/lib/mantid/lib/mantid/utils/_utils.py", line 46, in import_mantid_cext
    raise ImportError(msg)
ImportError: First import of "._api" failed with "/home/vagrant/.local/lib/mantid/lib/mantid/api/../../../lib/libMantidGeometry.so: undefined symbol: _ZN24BRepBuilderAPI_MakeShape5BuildEv". Second import of "_api" failed with "No module named '_api'"
```

I believe that error [comes from opencascade](https://forum.freecad.org/viewtopic.php?style=5&t=68128). Opencascade was added when we upgraded to f36 (and upgraded mantid versions). It has also caused problems by increasing the size of the beamsim image by 3gb. It is unclear to me, at the moment, what exactly opencascade is used for. But, my best guess is 3d visualization. Since we don't do any visualization with mantid (we just use the python api) I think it might be safe to remove it. There is a cmake option [to disable it](https://github.com/mantidproject/mantid/blob/v6.4.0/buildconfig/CMake/CommonSetup.cmake#L59) which I'll try.

### PR DESCRIPTION
**Pull request by e-carlin**
Remove opencascad-devel which brought in 3GB of dependencies. It caused the error on import, was large, and was unused because we don't do any rendering with Mantid. In addition, disable other unused parts of Mantid (ex docs).

Add ${module}_test which we can use for adding tests to modules which will hopefully prevent regressions like this from happening in the future.

### CODE REVIEW

File: installers/rpm-code/codes.sh [hunk 1]
```diff
@@ -208,44 +209,54 @@ codes_err() {
...
         "$p"
         codes_install_python_done
     fi
-    local d=${codes_dir[prefix]}/lib64
+    declare t=${module}_test
+    if codes_is_function "$t"; then
+        codes_msg "Running: $t"
+        # python adds cwd to PYTHONPATH so go to a path where
+        # cwd won't contain any python that would be added
+        mkdir -p /tmp/$(date -u +%Y%m%d.%H%M%S)
```
Comment (e-carlin) [inline installers/rpm-code/codes.sh:246]: @robnagler I want to call this out. Wasn't sure about a good location.

File: installers/rpm-code/codes.sh [hunk 2]
```diff
@@ -208,44 +209,54 @@ codes_err() {
...
+        cd "$_"
+        "$t"
+        cd -
```
Comment (robnagler) [inline installers/rpm-code/codes.sh:249]: This is not guaranteed to work since `$t` may cd.

File: installers/rpm-code/codes.sh (hunk 1 above)
Comment (robnagler) [reply installers/rpm-code/codes.sh:246]: Good idea. You can use install_tmp_dir which will clean it up better. Also it labels the directory a bit better so you know you can delete them manually. Finally, it uses /var/tmp which usually has more room than /tmp.

**Review by robnagler — CHANGES_REQUESTED** [review]

File: installers/rpm-code/codes.sh (hunk 1 above)
Comment (e-carlin) [reply installers/rpm-code/codes.sh]: 7c8960c

**Review by robnagler — APPROVED** [review]

**Pull request merged**